packages/kr.rb

```diff
@@ -0,0 +1,48 @@
+require 'package'
+
+class Kr < Package
+  description 'Krypton command-line client, daemon, and SSH integration'
+  homepage 'https://krypt.co'
+  version '2.4.4'
+  source_url 'https://raw.githubusercontent.com/kryptco/kr/master/README.md'
+  source_sha256 'a0452783d22218a6fab914abbf5263459aaa1c8745512d7c4ebcf51fe08aaf7a'
+
+  depends_on "gcc7" => :build
+  depends_on "rust" => :build
+  depends_on "go" => :build
+  depends_on "pkgconfig" => :build
+  depends_on "libsodium" => :build
+  depends_on "rsync" => :build
+  depends_on "buildessential" => :build
+  depends_on "psmisc" => :build
+  depends_on "git" => :build
+
+  def self.install
+    ENV["GOPATH"] = "#{Dir.pwd}/go"
+
+    system "rustup", "target", "add", "wasm32-unknown-emscripten"
+    system "cargo web --version || ( curl -Ls https://github.com/koute/cargo-web/archive/0.6.10.zip -o cargo-web.zip; unzip cargo-web.zip; cd cargo-web-0.6.10; cargo build --release && cargo install )"
+
+    system "mkdir -p #{Dir.pwd}/go/src"
+    system "go get github.com/kryptco/kr"
+    FileUtils.cd("go/src/github.com/kryptco/kr") do
+      ENV["PREFIX"] = "#{CREW_PREFIX}"
+      ENV["PWD"] = Dir.pwd
+      system "git checkout tags/2.4.4"
+      system "git submodule update --init --recursive"
+      system "git checkout -- Makefile"
+      system "sed -i '/^LINK_LIBSIGCHAIN_LDFLAGS/ s/$/ -fstack-protector/' Makefile"
+      system "sed -i '/^DSTBIN =/ s/.*/DSTBIN = #{CREW_DEST_PREFIX.gsub('/', '\/')}\\/bin/' Makefile"
+      system "sed -i '/^DSTLIB =/ s/.*/DSTLIB = #{CREW_DEST_PREFIX.gsub('/', '\/')}\\/lib/' Makefile"
+      system "make"
+      system "make", "install"
```

Comment:
Is this sufficient to install and track everything this package requires?  Just wondering if we're not tracking the rustup and cargo line stuff above.  In other words, does everything that gets installed get removed with `crew remove kr`?

Reply:
The answer is yes and no. Krypton and all of it's dependent files are tracked and removed, I have checked that multiple times, however some of the build environment stuff does not get tracked or removed. For instance cargo-web or the rustup target. I dont know of a way to handle that and I dont see any evidence of that ever being handled in the homebrew tap https://github.com/kryptco/homebrew-tap/blob/master/kr.rb or the aur package https://aur.archlinux.org/cgit/aur.git/tree/PKGBUILD?h=kr both of which I used to get this going.

Reply:
Can you specify a target directory for cargo and rustup?  If you install everything relative to `#{CREW_DEST_DIR}`, that may be the ticket.  By the way, just because homebrew and aur does it doesn't mean we have to do it the same way.  Package managers should do a better job of cleaning up the residue after an uninstall.

Reply:
Did some quick research and testing and now everything for rustup and cargo is sandboxed in the build directory and doesnt change anything for the user.

Reply:
So basically rustup and cargo stuff is only needed for the build?

Reply:
That I am aware of and can figure out, that is true.